merge_multi_operator_test.go

```diff
@@ -0,0 +1,82 @@
+package gorocksdb
+
+import (
+	"testing"
+
+	"github.com/facebookgo/ensure"
+)
+
+func TestMergeMultiOperator(t *testing.T) {
+	var (
+		givenKey     = []byte("hello")
+		startingVal  = []byte("foo")
+		mergeVal1    = []byte("bar")
+		mergeVal2    = []byte("baz")
+		fMergeResult = []byte("foobarbaz")
+		pMergeResult = []byte("barbaz")
+	)
+
+	merger := &mockMergeMultiOperator{
+		fullMerge: func(key, existingValue []byte, operands [][]byte) ([]byte, bool) {
+			ensure.DeepEqual(&fatalAsError{t}, key, givenKey)
+			ensure.DeepEqual(&fatalAsError{t}, existingValue, startingVal)
+			ensure.DeepEqual(&fatalAsError{t}, operands[0], pMergeResult)
+			return fMergeResult, true
+		},
+		partialMerge: func(key, leftOperand, rightOperand []byte) ([]byte, bool) {
```

Comment:
Do we have to implement both `MergeMultiOperator` and `MergeOperator` to make it work?

Reply:
Yes, unless we want to change the signature of https://github.com/DataDog/gorocksdb/blob/bibby/merge-multi/options.go#L143

Reply:
Ideally MergeOperator would only have `FullMerge` and then we could have a PartialMergeOperator and PartialMergeMultiOperator.    If it implemented neither of those, FullMerge would always be called...